Content.Server/Anomaly/Effects/EntityAnomalySystem.cs

```diff
@@ -22,55 +23,82 @@ public override void Initialize()
+                continue;
 
-        var xform = Transform(uid);
-        // A cluster of entities
-        SpawnEntitesOnOpenTiles(component, xform, component.MaxSpawnAmount, component.SpawnRange, component.Spawns);
-        // And so much meat (for the meat anomaly at least)
-        SpawnEntitesOnOpenTiles(component, xform, component.MaxSpawnAmount, component.SpawnRange, component.SuperCriticalSpawns);
+            SpawnEntitiesOnOpenTiles(component, entry, 1, 1);
+        }
     }
 
-    private void OnStabilityChanged(EntityUid uid, EntitySpawnAnomalyComponent component, ref AnomalyStabilityChangedEvent args)
+    private void OnStabilityChanged(Entity<EntitySpawnAnomalyComponent> component, ref AnomalyStabilityChangedEvent args)
     {
-        if (!component.SpawnOnStabilityChanged)
-            return;
+        foreach (var entry in component.Comp.Entries)
+        {
+            if (!entry.SpawnOnStabilityChanged)
+                continue;
 
-        var range = component.SpawnRange * args.Stability;
-        var amount = (int) (component.MaxSpawnAmount * args.Stability + 0.5f);
+            SpawnEntitiesOnOpenTiles(component, entry, args.Stability, args.Severity);
+        }
+    }
+
+    private void OnSeverityChanged(Entity<EntitySpawnAnomalyComponent> component, ref AnomalySeverityChangedEvent args)
+    {
+        foreach (var entry in component.Comp.Entries)
+        {
+            if (!entry.SpawnOnSeverityChanged)
+                continue;
 
-        var xform = Transform(uid);
-        SpawnEntitesOnOpenTiles(component, xform, amount, range, component.Spawns);
+            SpawnEntitiesOnOpenTiles(component, entry, args.Stability, args.Severity);
+        }
     }
 
-    private void SpawnEntitesOnOpenTiles(EntitySpawnAnomalyComponent component, TransformComponent xform, int amount, float radius, List<EntProtoId> spawns)
+    //TheShuEd:
+    //I know it's a shitcode! I didn't write it! I just restructured the functions
+    // To Do: make it reusable with TileAnomalySystem
+    private void SpawnEntitiesOnOpenTiles(Entity<EntitySpawnAnomalyComponent> component, EntitySpawnSettingsEntry entry, float stability, float severity)
     {
-        if (!component.Spawns.Any())
+        if (entry.Spawns.Count == 0)
             return;
 
-        if (!_map.TryGetGrid(xform.GridUid, out var grid))
+        var xform = Transform(component.Owner);
+        if (!TryComp<MapGridComponent>(xform.GridUid, out var grid))
             return;
 
+        var amount = (int) (MathHelper.Lerp(entry.MinAmount, entry.MaxAmount, severity * stability) + 0.5f);
+
         var localpos = xform.Coordinates.Position;
         var tilerefs = grid.GetLocalTilesIntersecting(
-            new Box2(localpos + new Vector2(-radius, -radius), localpos + new Vector2(radius, radius))).ToArray();
+            new Box2(localpos + new Vector2(-entry.MaxRange, -entry.MaxRange), localpos + new Vector2(entry.MaxRange, entry.MaxRange))).ToArray();
 
```

Comment:
Use a List on the system or the likes as this will keep making array allocs.

Entites should also be Entities.

Also the method should be combined with the tile one as mentioned.